Signal/src/call/CallService.swift

```diff
@@ -122,85 +224,86 @@ protocol CallServiceObserver: class {
                 }
                 stopAnyCallTimer()
-                if let call = call {
-                    DeviceSleepManager.sharedInstance.addBlock(blockObject: call)
+                if let callData = callData {
+                    DeviceSleepManager.sharedInstance.addBlock(blockObject: callData)
                     self.startCallTimer()
                 }
             }
 
-            Logger.debug("\(self.logTag) .call setter: \(oldValue?.identifiersForLogs as Optional) -> \(call?.identifiersForLogs as Optional)")
+            Logger.debug("\(self.logTag) .callData setter: \(oldValue?.call.identifiersForLogs as Optional) -> \(callData?.call.identifiersForLogs as Optional)")
 
             for observer in observers {
-                observer.value?.didUpdateCall(call: call)
+                observer.value?.didUpdateCall(call: callData?.call)
             }
         }
     }
 
-    // Used to coordinate promises across delegate methods
-    private var fulfillCallConnectedPromise: (() -> Void)?
-    private var rejectCallConnectedPromise: ((Error) -> Void)?
-
-    // Used by waitForPeerConnectionClient to make sure any received
-    // ICE messages wait until the peer connection client is set up.
-    private var fulfillPeerConnectionClientPromise: (() -> Void)?
-    private var rejectPeerConnectionClientPromise: ((Error) -> Void)?
-    private var peerConnectionClientPromise: Promise<Void>?
+    // TODO: Remove?
+    var call: SignalCall? {
+        get {
+            SwiftAssertIsOnMainThread(#function)
 
-    // Used by waituntilReadyToSendIceUpdates to make sure CallOffer was 
-    // sent before sending any ICE updates.
-    private var fulfillReadyToSendIceUpdatesPromise: (() -> Void)?
-    private var rejectReadyToSendIceUpdatesPromise: ((Error) -> Void)?
-    private var readyToSendIceUpdatesPromise: Promise<Void>?
+            guard let callData = callData else {
+                return nil
+            }
+            return callData.call
+        }
+    }
 
-    weak var localVideoTrack: RTCVideoTrack? {
-        didSet {
+    var peerConnectionClient: PeerConnectionClient? {
+        get {
             SwiftAssertIsOnMainThread(#function)
 
-            Logger.info("\(self.logTag) \(#function)")
-
-            fireDidUpdateVideoTracks()
+            guard let callData = callData else {
+            return nil
+            }
+            return callData.peerConnectionClient
         }
     }
 
-    weak var remoteVideoTrack: RTCVideoTrack? {
-        didSet {
+    var localVideoTrack: RTCVideoTrack? {
+        get {
             SwiftAssertIsOnMainThread(#function)
 
-            Logger.info("\(self.logTag) \(#function)")
-
-            fireDidUpdateVideoTracks()
+            guard let callData = callData else {
+                return nil
+            }
+            return callData.localVideoTrack
         }
     }
-    var isRemoteVideoEnabled = false {
-        didSet {
+    weak var remoteVideoTrack: RTCVideoTrack? {
```

Comment:
I think this `weak` was copied over - now that videoTrack lives on the callData (which I think is a good call!) we don't need it.

Reply:
Removed.